crypto/compactcert/committableSignatureSlot.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package compactcert
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+type committableSignatureSlot struct {
+	sigCommit           sigslotCommit
+	serializedSignature []byte
+	isEmptySlot         bool
+}
+
+const (
+	// ErrIndexOutOfBound returned when an index is out of the array's bound
+	ErrIndexOutOfBound = "pos %d past end %d"
```

Comment:
nit: There's another version of this string here: [crypto/merklesignature/committablePublicKeys.go](https://github.com/algorand/go-algorand/pull/2990/files#diff-bad406866bae28f1deac671cd7992227ad06e9945554853e250518bc3aa1d45f)

Maybe they are unloaded in Github, but I don't see them used anywhere. Could they be private?

:upside_down_face: this could be a struct with position / length, and format the message with `Error()`. That might allow using `errors.Is` if we do check for the error elsewhere.

Reply:
They should be exported for the user of the function to be able to check the error type, right?
I'll definitely change it to a proper error instead of string though in both places